spec.md

```diff
@@ -499,6 +499,54 @@ the module.  The `c`{.firrtl} sub-field contained in the `b`{.firrtl} sub-field
 flows into the module, and the `d`{.firrtl} sub-field contained in the
 `b`{.firrtl} sub-field flows out of the module.
 
+## Type Modifiers
+
+### Constant Type
+
+A constant type is a type whose value is guaranteed to be a compile-time 
+constant.  Constant types may be used in ports, wire, nodes, and generally 
+anywhere a non-constant type is usable.  Operations on constant type are well 
+defined.  As a general rule (with any exception listed in the definition for 
+such operations as have exceptions), an operation whose arguments are constant
+produces a constant.  An operations with some non-constant arguments produce a
+non-constant.  Constants can be used in any context with a source flow which 
+allows a non-constant.  Constants may be used as the target of a connect so long 
+as the source of the connect is itself constant.  These rules ensure all 
+constants are derived from literals or from constant-typed input ports of the 
+top-level module.
+
+``` firrtl
+const UInt<3>
```

Comment:
shouldn't this result in something in the parser grammar at the end?

Reply:
thanks.